controller/Lists.php

```diff
@@ -292,24 +295,32 @@ public function saveLists(ValueCollectionService $valueCollectionService): void
             )
         );
 
-        foreach ($payload as $key => $value) {
-            if (preg_match('/^list-element_/', $key)) {
-                $encodedUri = preg_replace('/^list-element_[0-9]+_/', '', $key);
-                $uri = tao_helpers_Uri::decode($encodedUri);
-                $newUriValue = trim($payload["uri_$key"] ?? '');
-                $element = $elements->extractValueByUri($uri);
+        $listElements = array_filter($payload, function (string $key) {
+            return (bool)preg_match('/^list-element_/', $key);
+        },
+        ARRAY_FILTER_USE_KEY);
 
-                if ($element === null) {
-                    $elements->addValue(new Value(null, $newUriValue, $value));
+        $maxItems = $this->getListService()->getMaxItems();
+        if ($maxItems > 0) {
+            $listElements = array_slice($listElements, 0, $maxItems);
+        }
```

Comment:
Do you think we should truncated? 

It there another way to let the user know they are trying to cross the limits? 

Do you think `$valueCollectionService->persist(` should validate this?

Reply:
Moved validation to the service in 71b81f0fa0c853eedd4e7ab54a11a1829f06dde5.

In order to do so, we need to merge this pull request in tao-core first: https://github.com/oat-sa/tao-core/pull/3303/files

Now when max number of values is exceeded, we throw an error which is shown as an error message in the frontend when the user attempts to save the list. No item is persisted by `ValueCollectionService::persist()` (tao-core) in that case.